### PR TITLE
feat: removing closable when there is no background (#558)

### PR DESCRIPTION
**Pull request by Lucassec1**
## Description

Now, whenever the alert has no background, it cannot receive the closable.

## Screenshots

image

### CODE REVIEW

File: projects/ion/src/lib/alert/alert.component.ts [hunk 1]
```diff
@@ -35,4 +43,10 @@ export class IonAlertComponent implements OnInit {
   ngOnInit(): void {
     this.setIcon();
   }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.hideBackground && changes.hideBackground.currentValue) {
+      this.closable = false;
+    }
+  }
```
Comment (iurynogueira) [inline projects/ion/src/lib/alert/alert.component.ts:51]: I believe that you can add this rule in OnInit.

File: projects/ion/src/lib/alert/alert.component.ts [hunk 2]
```diff
@@ -35,4 +43,10 @@ export class IonAlertComponent implements OnInit {
   ngOnInit(): void {
     this.setIcon();
```
Comment (iurynogueira) [inline projects/ion/src/lib/alert/alert.component.ts:44]:
```suggestion
    this.setIcon();
    
    if (check is hide bg) {
      this.closable = false;
    }
```

**Review by iurynogueira — CHANGES_REQUESTED** [review]

**Comment by AlyssonMascarenhas** [comment]
Check conflicts

File: projects/ion/src/lib/alert/alert.component.ts (hunk 1 above)
Comment (Lucassec1) [reply projects/ion/src/lib/alert/alert.component.ts:51]: I added the code to the ngOnInit() method, but when I click on the hideBackground button in the Storybook, the closable property remains true.

**Review by AlyssonMascarenhas — APPROVED** [review]

**Review by iurynogueira — APPROVED** [review]

**Pull request merged**